assignments/callbacks.js

```diff
@@ -23,27 +25,61 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
   });
 
 */
-
-
-function getLength(arr, cb) {
-  // getLength passes the length of the array into the callback.
+// HERE IS THE ARRAY FOR THE PROBLEMS AS WELL AS THE HOF
+const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
+function itemsHOF (arr, cb){
+  return cb(arr);
 }
 
-function last(arr, cb) {
-  // last passes the last item of the array into the callback.
+// function getLength(arr, cb) {
+//   // getLength passes the length of the array into the callback.
+// }
+//Answer One
+function itemsLength (arr) {
+  return arr.length;
 }
 
-function sumNums(x, y, cb) {
-  // sumNums adds two numbers (x, y) and passes the result to the callback.
+// function last(arr, cb) {
+//   // last passes the last item of the array into the callback.
+// }
+// Answer Two
+function lastItem (arr) {
+  return arr.slice(-1);
 }
 
-function multiplyNums(x, y, cb) {
-  // multiplyNums multiplies two numbers and passes the result to the callback.
+// function contains(item, list, cb) {
+//   // contains checks if an item is present inside of the given array/list.
+//   // Pass true to the callback if it is, otherwise pass false.
+// }
+// Answer three
+function itemsHOF2 (array, callback, item){
+  return callback(array, item);
+};
+function checkItem(array, item) {
+  return array.includes(item);
+};
+
+// END ITEMS ARRAY PROBLEMS HERE
+// BEGIN MATH ARRAY PROBLEMS HERE
+
+// HERE IS THE MATH HOF
+function MathCalcuator(num1, num2, cb){
+  return cb(num1, num2);
+}
+// function sumNums(x, y, cb) {
+//   // sumNums adds two numbers (x, y) and passes the result to the callback.
+// }
+// Answer One
+function sumMath(num1, num2) {
+  return num1 + num2;
 }
 
-function contains(item, list, cb) {
-  // contains checks if an item is present inside of the given array/list.
-  // Pass true to the callback if it is, otherwise pass false.
+// function multiplyNums(x, y, cb) {
+//   // multiplyNums multiplies two numbers and passes the result to the callback.
+// }
+// Answer Two
+function proMath(num1, num2) {
+  return num1 * num2;
 }
 
 /* STRETCH PROBLEM */
```

Comment:
You went about doing yours a bit different than the instructions but I think you understand the concept! ✔️